powertools-core/src/main/java/software/amazon/lambda/powertools/core/internal/UserAgentConfigurer.java

```diff
@@ -0,0 +1,90 @@
+package software.amazon.lambda.powertools.core.internal;
+
+import com.sun.org.slf4j.internal.Logger;
+import com.sun.org.slf4j.internal.LoggerFactory;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.URL;
+import java.util.Properties;
+
+import static software.amazon.lambda.powertools.core.internal.SystemWrapper.getenv;
+
+public class UserAgentConfigurer {
+
+    public static final String NA = "NA";
+    public static final String VERSION_KEY = "powertools.version";
+    public static final String PT_FEATURE_VARIABLE = "${PT_FEATURE}";
+    public static final String PT_EXEC_ENV_VARIABLE = "${PT_EXEC_ENV}";
+    public static final String VERSION_PROPERTIES_FILENAME = "version.properties";
+    public static final String AWS_EXECUTION_ENV = "AWS_EXECUTION_ENV";
+    private static final Logger LOG = LoggerFactory.getLogger(UserAgentConfigurer.class);
+    private static final String NO_OP = "no-op";
+    private static String PT_VERSION = getProjectVersion();
+    private static String USER_AGENT_PATTERN = "PT/" + PT_FEATURE_VARIABLE + "/" + PT_VERSION + " PTEnv/" + PT_EXEC_ENV_VARIABLE;
+
+    /**
+     * Retrieves the project version from the version.properties file
+     *
+     * @return the project version
+     */
+    static String getProjectVersion() {
```

Comment:
Can this be private?

Reply:
this for test purposes left to package access. It can get moved to a dedicated package though..

Reply:
Nah, I think that's reasonable as it is!